### PR TITLE
Allow namespaced configs for blueprints in jdl.

### PR DESCRIPTION
**Pull request by mshima**
Any kind of deterministic option parser for blueprints can make projects like jdl-studio more complex.
Or we must centralize all blueprints options and values here at main generator.

This proposal implements namespaced configs for blueprints.

Blueprint config(`foo`) requires `foo` to be in blueprints.

```
application {
  config {
    baseName jhipster
    blueprints [foo]
   }
  config(foo) {
    fooConfig fooConfigValue
  }
}
```

results in: 
```
{
  "generator-jhipster": {
    "baseName": "jhipster",
    "blueprints": [
      {
        "name": "generator-jhipster-foo"
      }
    ]
  },
  "namespaceConfigs": {
    "generator-jhipster-foo": {
      "fooConfig": "fooConfigValue"
    }
  }
}
```

Fixes https://github.com/jhipster/generator-jhipster/issues/23959, https://github.com/jhipster/generator-jhipster/issues/9870

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: jdl/validators/enum-validator.spec.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ describe('jdl - EnumValidator', () => {
       context('with a reserved class name as name', () => {
         it('should fail', () => {
           expect(() => {
-            validator.validate(new JDLEnum({ name: 'Catch' }));
+            validator.validateBusiness(new JDLEnum({ name: 'Catch' }));
```
Comment (mraible) [inline jdl/validators/enum-validator.spec.ts:59]: Why did this method name need to change?

Comment (mshima) [reply jdl/validators/enum-validator.spec.ts:59]: I've split basic jdl validation and business validation.

Currently if blueprints config is passed, all validation is skipped:
https://github.com/jhipster/generator-jhipster/blob/da7bc6f17a99f206112a89886af43cf5db1c6fcd/jdl/validators/jdl-with-application-validator.ts#L49-L59

Now only business validation are ignored:
https://github.com/jhipster/generator-jhipster/blob/69e8c39793d6a02a302880cc3fda3a343af8eb4a/jdl/validators/jdl-with-application-validator.ts#L51-L65

Comment (mshima) [reply jdl/validators/enum-validator.spec.ts:59]: I will add options instead of splitting.
It’s cleaner.

Comment (mshima) [reply jdl/validators/enum-validator.spec.ts]: I suppose this is resolved

File: generators/jdl/generator.spec.mts [hunk 2]
```diff
@@ -68,7 +68,7 @@ describe(`generator - ${generator}`, () => {
           inline: 'entity Foo {}',
           baseName: 'jhipster',
         }),
-      ).rejects.toThrow('Database type is required to validate entities.');
+      ).rejects.toThrow("The JDL object, the application's name and its the database type are mandatory.");
```
Comment (mraible) [inline generators/jdl/generator.spec.mts:71]: Oxford comma: I'd add a comma after name in this sentence. Feel free to ignore.

**Review by mraible — DISMISSED** [review]

**Comment by mraible** [comment]
@gourav Does this syntax work for you?

**Comment by mshima** [comment]
For entity-audit this should work:
```jdl
application {
  config {
    applicationType monolith
    blueprints [entity-audit]
    entity-audit:auditFramework javers
  }
  entity *
}

@EnableAudit
entity Foo {}
```

`auditUpdateType` is a runtime option for cli and prompt designed when it was a module, executed after the app generation. It won't be available as jdl config.
If the idea is to opt-out audit instead of opt-in a new config should be added.

**Review by DanielFran — APPROVED** [review]

**Comment by gourav** [comment]
@mraible @mshima Yes. Syntax would work.
Although to be consistent with how blueprints are designed, it may be good idea to introduce blueprintConfig key and move all blueprints specific options under it ?

Cuurent syntax would work too.

**Comment by mshima** [comment]
The following syntax would work too.
```
application {
  config {
    applicationType monolith
    blueprints [entity-audit]
  }

  config:entity-audit {
    auditFramework javers
  }

  entity *
}
```

JDL parsing PR would be more complex.

**Comment by DanielFran** [comment]
> The following syntax would work too.
> 
> ```
> application {
>   config {
>     applicationType monolith
>     blueprints [entity-audit]
>   }
> 
>   config:entity-audit {
>     auditFramework javers
>   }
> 
>   entity *
> }
> ```

I prefer this one!

File: jdl/models/jdl-application-configuration-factory.ts [hunk 3]
```diff
@@ -31,17 +31,50 @@ export default function createApplicationConfigurationFromObject(configurationOb
   const configuration = new JDLApplicationConfiguration();
   Object.keys(configurationObject).forEach(optionName => {
     const optionValue = configurationObject[optionName];
-    if (!applicationDefinition.doesOptionExist(optionName)) {
+    if (!applicationDefinition.doesOptionExist(optionName) && !optionName.includes(':')) {
```
Comment (DanielFran) [inline jdl/models/jdl-application-configuration-factory.ts:34]: Is this still needed?

Comment (mshima) [reply jdl/models/jdl-application-configuration-factory.ts:34]:
```suggestion
    if (!applicationDefinition.doesOptionExist(optionName)) {
```

**Review by DanielFran — APPROVED** [review]

**Comment by DanielFran** [comment]
@mshima Seems much better now!
Documentation will need updates to include this new configuration

@deepu105 I suppose JDL Studio needs to be updated accordingly?

**Comment by deepu105** [comment]
For JDL just updating to new version of JHipster should be enough unless the syntax is brand new

**Pull request merged**